tests/unit/common/test_table.py

```diff
@@ -1401,6 +1402,18 @@ def setUpContainer(self):
         return table
 
 
+class TestDynamicTableAddEnumRoundTrip(H5RoundTripMixin, TestCase):
+
+    def setUpContainer(self):
+        table = DynamicTable('table0', 'an example table')
+        table.add_column('bar', 'an enumerable column', enum=True)
+        table.add_row(bar='a')
+        table.add_row(bar='b')
+        table.add_row(bar='a')
+        table.add_row(bar='c')
```

Comment:
Can you make a separate non-round-trip test with this code and test for the expected result? And do the same for an indexed enum column? What is the expected result? I ran:
```
>>> table = DynamicTable('table0', 'an example table')
>>> table.add_column('foo', 'an indexed enumerable column', enum=True, index=True)
>>> table.add_row(foo=['a', 'b'])
>>> table.add_row(foo=['a', 'b'])
>>> table.add_row(foo=['a', 'b', 'c'])
>>> table.foo
<hdmf.common.table.EnumData object at 0x0000016EA474E708>
>>> table.foo.data
['a', 'b', 'a', 'b', 'a', 'b', 'c']
>>> table.foo.elements
<hdmf.common.table.VectorData object at 0x0000016EA474E948>
>>> table.foo.elements.data
[]
>>> table['foo']
<hdmf.common.table.VectorIndex object at 0x0000016EA474EA48>
>>> table['foo'].data
[2, 4, 7]
```
which seems a little counter intuitive. I would think either each element of the row should be enumerated, or the array for the row should be enumerated, but here, nothing is enumerated.

This can be done in a separate PR and since it is experimental, it is not critical to get it in the next release.

Reply:
Can you add non-round-trip tests for a non-indexed enum column and an indexed enum column? I'm also happy to do it.

Reply:
I'm not sure what you're asking for... Does the TestCase starting at line 1153 work for the non-index enum column?

Reply:
Yeah, the TestCase at line 1153 works for the non-index enum column because `DynamicTable.add_row` just calls `EnumData.add_row`. So that's fine. My bad. The indexed enum column is more complicated though and should be tested separately.